Upload file

ael-code opened the pull request:
Please look at the commit messages

ael-code commented:
I am ashamed of my previous version of logging system. Please excuse me ;)

boyska commented:
mh, I actually liked it more in the previous version. You pointed out that this was the way suggested by flask. I'm not sure, I'd like some more explanation; plus, I want to reproduce and understand the bug that you found on my pull request.

ael-code commented:
#### About the bug:
I spent some time to understand the bug and solve it.
I am quite sure that if you see fsdb debug messages running your latest code you probably have something misconfigured.
The problem was caused by the fact that on the first call of initLoggers we were setting the log level on both logger and handler object but on succeding calls we only changed the level of the logger object.

#### About the logger reorganization:
The previous version had a lot of problems:
 - you had to initialize all the logger every time. For example ```libreant-manager``` do not use agherant but it initialize the relative logger.
 - the code was not suit for running multiple time. Handler object was recreated at every call.
 - the code was not easy to maintain. If some sub-module change its logger name or you want to add a sub-module somewhere in your code you have to change also the initLogger function.

Flask in [logging section](http://flask.pocoo.org/docs/0.10/errorhandling/#other-libraries) of its docs suggest to use the multiple getLogger solution in order to have the the possibility to use different loggers/handlers for different modules, but I think that in the near future we won't use this chance.

The new code do not suffer from problems pointed before but f you mind we'll need

boyska commented:
ok, this seems a good explanation. I'm still a bit in favor of explicit mentioning what we need, but well, if there is no practical disadvantage, your solution is much simpler.
I'll test it soon

boyska commented:
This is as bad as I thought. It will open debug for **everything**, including elasticsearch. Which means that you'll get every message exchanged from/to the elasticsearch, resulting in an incredibly huge amount of logging. I don't that this is desirable. I still believe that explicitly hardcoding what do we want to debug is much better. Then we can discuss about overriding this with something completely flexible ( like [logging.config.dictConfig](https://docs.python.org/2/library/logging.config.html#logging.config.dictConfig) ), but if we want to provide a default, this doesn't look to me like a good default.

ael-code commented:
Ok I have tried it without starting elasticsearch so I did not realized that. Sorry.

logging.config.dictConfig seems a good choice. Are you going to try it?

boyska commented:
@ael-code , I think that implementing `dictConfig` is a very good thing to do, but I don't think it is very urgent.

ael-code committed:
cli libreant-db: make METADATA optional

ael-code committed:
bugfix: BadOptionUsage accepts only one string on click==[6.0, 6.2]

additional infos can be found here:
    https://github.com/mitsuhiko/click/pull/475
